### ISSUE TITLE
[Bug]: Index Out of Bounds for PNG with Transparent Background

### ISSUE THREAD

**Issue by YannickAlex07** [YannickAlex07/imghash-rs#12]
### Version

1.1.1

### Describe the Bug

While using any image with a transparent background, the crate seems to produce an `Index Out of Bounds`-error for the average hash. The part that seems to be producing this issues is this code:

```rust
let mut bits = vec![false; (self.width * self.height) as usize];
for (i, p) in converted.as_bytes().to_vec().iter().enumerate() {
    if *p as usize > mean {
        bits[i] = true;
    }
}
```

This happens because the converted image actually contains double the amount of data compared to its size. My initial assumption is that each pixel results in two values here => one for the grayscale value and one for the transparency value.

To make sure the crate stays somewhat compatible with some Python packages, it would make sense to check how PIL handles this case and how the resulting data looks like.

### On which platform did you observe this bug?

MacOS

**Comment by YannickAlex07** [comment]
If the grayscaling is indeed the issue, this could also be resolved by implementing our own grayscaling which is needed for task #10.

**Comment by YannickAlex07** [comment]
Should be resolved with #13 - however needs some further testing before it will be closed
